prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -119,9 +109,41 @@ class ReportFunction {
         actionHistory.trackActionResult(httpResponseMessage)
         workflowEngine.recordAction(actionHistory)
         actionHistory.queueMessages() // Must be done after creating TASK record.
+        // write the data to the table if we're dealing with covid-19. this has to happen
+        // here AFTER we've written the report to the DB
+        writeCovidResultMetadataForReport(report, context, workflowEngine)
         return httpResponseMessage
     }
 
+    private fun writeCovidResultMetadataForReport(
+        report: Report?,
+        context: ExecutionContext,
+        workflowEngine: WorkflowEngine
+    ) {
+        if (report != null && report.schema.topic.lowercase() == "covid-19") {
+            // next check that we're dealing with an external file
+            val clientSource = report.sources.firstOrNull { it is ClientSource }
```

Comment:
So, we only store metadata about the initial incoming file?   That seems incomplete.   For example, Matt's download system needs data about rows in outdoing Reports.

(At one time, I was considering getting rid of these Sources as they didn't seem very useful.)

Reply:
Currently yes. Adding downloads would duplicate all the data. My sense was we could get this data from the report lineage for files, something like joining on the message_id and the incoming file(s) that fed an outbound report. If that's too expensive we could add a flag to the table for inbound/outbound to denormalize it a little.